task_02/danil_dyachenko/02_knight_and_princess/graph_gen.cpp

```diff
@@ -0,0 +1,107 @@
+#include <iostream>
+#include <fstream>
+#include <vector>
+
+struct Vertex{
+    int id;
+    std::vector<int> edges;
+};
+
+struct Edge{
+    int id;
+    std::vector<int> vertices;
+};
+
+class Graph{
+    std::vector<Vertex> vertices;
+    std::vector<Edge> edges;
+public:
+    void add_vertices(int n_vertices = 1){
+        int n_edges = 1;
+        int x;
+        vertices.resize(n_vertices);
+        for (int i = 0; i < n_vertices; ++i){
+            std::cout << "n_edges for " << i << " vertex: ";
+            std::cin >> n_edges;
+            vertices[i].edges.resize(n_edges);
+            vertices[i].id = i;
+            for (int j = 0; j < n_edges; ++j){
+                std::cin >> x;
+                vertices[i].edges[j] = x;
+            }
+        }
+    }
+    void print_vertices(void){
```

Comment:
Зачем здесь `void`?
```suggestion
    void print_vertices() {
```

Reply:
Если функция не мутирует родительский объект, тогда она должна быть помечена как `const`
```suggestion
    void print_vertices() const {
```